VideoWeb/VideoWeb/ClientApp/src/app/shared/beta-banner/beta-banner.component.ts

```diff
@@ -1,56 +1,61 @@
-import { Component, OnInit, Input } from '@angular/core';
-import { Router, NavigationEnd } from '@angular/router';
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { NavigationEnd, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
 import { filter } from 'rxjs/operators';
 import { ConferenceStatus } from 'src/app/services/clients/api-client';
 import { EventsService } from 'src/app/services/events.service';
-import { AdalService } from 'adal-angular4';
 
 @Component({
-  selector: 'app-beta-banner',
-  templateUrl: './beta-banner.component.html'
+    selector: 'app-beta-banner',
+    templateUrl: './beta-banner.component.html'
 })
-export class BetaBannerComponent implements OnInit {
-  pageUrl: string;
-  readonly inPageFeedbackUrl = 'https://www.smartsurvey.co.uk/s/VideoHearings_Feedback/?pageurl=';
-  readonly exitSurveyUrl = 'https://www.smartsurvey.co.uk/s/VideoHearings_ExitSurvey/?pageurl=';
-  @Input() isRepresentativeOrIndividual: boolean;
-  status: ConferenceStatus;
+export class BetaBannerComponent implements OnInit, OnDestroy {
+    pageUrl: string;
+    readonly inPageFeedbackUrl = 'https://www.smartsurvey.co.uk/s/VideoHearings_Feedback/?pageurl=';
+    readonly exitSurveyUrl = 'https://www.smartsurvey.co.uk/s/VideoHearings_ExitSurvey/?pageurl=';
+    @Input() isRepresentativeOrIndividual: boolean;
+    status: ConferenceStatus;
 
-  constructor(
-    private router: Router,
-    private eventService: EventsService) {
-    this.pageUrl = '';
-  }
-  ngOnInit(): void {
-    this.getInPageFeedbackUrl();
-    this.setupSubscribers();
-  }
-  private setupSubscribers() {
-    this.eventService.start();
-    this.eventService.getHearingStatusMessage().subscribe(message => {
-      this.status = message.status;
-      if (message.status === ConferenceStatus.Closed) {
-        this.updateFeedbackUrl(this.exitSurveyUrl);
-      } else {
-        this.updateFeedbackUrl(this.inPageFeedbackUrl);
-      }
-    });
-  }
+    private subscriptions = new Subscription();
 
-  private getInPageFeedbackUrl() {
-    this.router.events
-      .pipe(filter(event => event instanceof NavigationEnd))
-      .subscribe((event: NavigationEnd) => {
+    constructor(private router: Router, private eventService: EventsService) {
         this.pageUrl = '';
-        if (this.status === ConferenceStatus.Closed && (this.router.url.indexOf('waiting-room') > -1)) {
-          this.updateFeedbackUrl(this.exitSurveyUrl);
-        } else {
-          this.updateFeedbackUrl(this.inPageFeedbackUrl);
-        }
-      });
-  }
+    }
+    ngOnInit(): void {
+        this.getInPageFeedbackUrl();
+        this.setupSubscribers();
+    }
 
-  updateFeedbackUrl(feedbackUrl: string): void {
-    this.pageUrl = feedbackUrl + this.router.url;
-  }
+    ngOnDestroy(): void {
```

Comment:
Why are we doing this in beta banner?

Reply:
we need to manage our subscriptions. I noticed this when i was digging through tests calling the API. If we don't unsubscribe between components then we will get memory leaks. We should either convert these api calls to promises or unsubscribe when the component is destroyed